homework/src/index.js

```diff
@@ -30,18 +29,74 @@
-      } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
+  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
+  const root = document.getElementById('root');
+  createAndAppend("img", root, { src: "./hyf.png", id: "imgLogo" });
+  const divSelect = createAndAppend("div", root, { id: "divSelect" });
+  createAndAppend("p", divSelect, { html: "HYF Repositories : ", id: "p1" });
+  const select = createAndAppend("select", divSelect, { id: "select" });
+  const divFlex = createAndAppend("div", root, { id: "divFlex" });
+  const divInfo = createAndAppend("div", divFlex, { id: "divInfo" });
+  const divCont = createAndAppend("div", divFlex, { id: "divCont" });
+  const divError = createAndAppend("div", root, { id: "divError" });
+
+
+  function main(HYF_REPOS_URL) {
+    const myCallback = (error, repositories) => {
+      if (error) {
+        divError.innerHTML = "";
+        divError.innerHTML = error;
+        return;
       }
-    });
+
+      repositories.forEach((repository, index) => {
+        createAndAppend("option", select, { html: repository.name, value: index });
+      });
+      select.addEventListener("change", (event) => {
+        renderRepository(divInfo, repositories[event.target.value]);
+        renderContributors(divCont, repositories[event.target.value]);
+      });
+      renderRepository(divInfo, repositories[0]);
+      renderContributors(divCont, repositories[0]);
+    }
+    fetchJSON(HYF_REPOS_URL, myCallback);
   }
 
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
+  function renderRepository(divInfo, repository) {
+    divInfo.innerHTML = "";
+    const table = createAndAppend("table", divInfo);
+    const tBody = createAndAppend("tBody", table);
+    const tr1 = createAndAppend("tr", tBody);
+    createAndAppend("td", tr1, { html: "Repository : " });
+    const link = createAndAppend('a', tr1, { html: repository.name });
+    link.setAttribute('href', repository.html_url);
+    const tr2 = createAndAppend("tr", tBody);
+    createAndAppend("td", tr2, { html: "Description : " });
+    createAndAppend("td", tr2, { html: repository.description });
+    const tr3 = createAndAppend("tr", tBody);
+    createAndAppend("td", tr3, { html: "Forks : " });
+    createAndAppend("td", tr3, { html: repository.forks });
+    const tr4 = createAndAppend("tr", tBody);
+    createAndAppend("td", tr4, { html: "Updated : " });
+    createAndAppend("td", tr4, { html: repository.updated_at.toLocaleString() });
+  }
 
+  function renderContributors(divCont, repository) {
+    divCont.innerHTML = "";
+    createAndAppend('h2', divCont, { html: "Contributions", class: "h2" });
+
+    fetchJSON(repository.contributors_url, (error, contributors) => {
+      if (error) {
+        createAndAppend('div', root, { html: error.message, class: 'alert-error' });
```

Comment:
there is nothing preventing the function from to stop if an error happen!! 
so the next line going to execute in any case which will make a new error which may crash your client functionalities.

you can solve this only by `return` to stop your whole function if error accords.

Reply:
Corrected.